lib/azure/armrest/virtual_machine_service.rb

```diff
@@ -120,12 +120,102 @@ def stop(vmname, group = armrest_configuration.resource_group)
         vm_operate('powerOff', vmname, group)
       end
 
+      # Update Virtual Machine
+      # location is required
+      # parameters example
+      #  data = {
+      #    location:"westeurope",
+      #    properties: {
+      #      storageProfile: {
+      #        dataDisks: [
+      #          {
+      #            name:"diskname",
+      #            diskSizeGB:"1",
+      #            lun: 0,
+      #            vhd: {
+      #              uri:"https://myaccount.blob.core.windows.net/vhds/vhdname.vhd"
+      #            },
+      #            createOption: "Empty"
+      #          }
+      #        ]
+      #      }
+      #    }
+      #  }
+      #
+      def update(vmname, data, rgroup = armrest_configuration.resource_group)
```

Comment:
The base `ResourceGroupBasedService` already has a `update` method (an alias to `create`), and your implementation is the same. This implementation is thus unnecessary. Besides your return is wrong.